FusionIIIT/applications/otheracademic/views.py

```diff
@@ -72,8 +147,166 @@ def reject_leave(request, leave_id):
+    receiver_value_designation = HoldsDesignation.objects.filter(user=receiver_value)
+    lis = list(receiver_value_designation)
+    obj = lis[0].designation
+    leave_entry=get_object_or_404(LeavePG,id=leave_id)
+
+        
+    file_id_forward= create_file(
+            uploader=request.user.username,
+            uploader_designation=obj,
+            receiver=leave_entry.hod,
+            receiver_designation="student",
+            src_module="otheracademic",
+            src_object_id=leave_id,
+            file_extra_JSON={"value": 2},
+            attached_file=None,
+            subject='pg_leave')
+    message = "A new leave application"
+    hod_user=User.objects.get(username=leave_entry.hod)
+    otheracademic_notif(request.user,hod_user, 'alert', leave_id, 'student', message)
+
+    return redirect('/otheracademic/leaveApproveTA')  # Redirect to appropriate page after approval
+
+def reject_leave_ta(request, leave_id):
+    leave_entry = LeavePG.objects.get(id=leave_id)
+    leave_entry.ta_rejected = True
+    leave_entry.save()
+    return redirect('/otheracademic/leaveApproveTA')  # Redirect to appropriate page after rejection
+
+def leaveApproveHOD(request):
+    
+    inbox=view_inbox(username=request.user.username,designation="student",src_module="otheracademic")
+    leave_ids=[msg ['src_object_id'] for msg in inbox if  msg['subject']=='pg_leave']
+    
+    form_data = LeavePG.objects.filter(id__in=leave_ids)
+    return render(request, 'otheracademic/leaveApproveHOD.html', {'form_data': form_data})
+
+def approve_leave_hod(request, leave_id):
+    leave_entry = LeavePG.objects.get(id=leave_id)
+    leave_entry.hod_approved = True
+    leave_entry.save()
+    return redirect('/otheracademic/leaveApproveHOD')  # Redirect to appropriate page after approval
+
+def reject_leave_hod(request, leave_id):
+    leave_entry = LeavePG.objects.get(id=leave_id)
+    leave_entry.hod_rejected = True
+    leave_entry.save()
+    return redirect('/otheracademic/leaveApproveHOD')  # Redirect to appropriate page after rejection
+
+
+def leaveStatusPG(request):
+    
+    form_data = LeavePG.objects.all()
+    roll_no = request.user.username
+    print(roll_no)
+    # form_data = LeavePG.objects.all()
+    return render(request, 'otheracademic/leaveStatusPG.html', {'form_data': form_data, 'roll_no' : roll_no})
+
+
+
+def leaveStatusPG_Dip(request):
+    inbox=view_inbox(username=request.user.username,designation="student",src_module="otheracademic")
+    leave_ids=[msg ['src_object_id'] for msg in inbox if  msg['subject']=='pg_leave']
+
+    form_data = LeavePG.objects.filter(id__in=leave_ids)
+    return render(request, 'otheracademic/leaveStatusPG_Dip.html', {'form_data': form_data})
+
+
 def graduateseminar(request):
-    return render(request,'otheracademic/graduateseminarForm.html')
+    user=get_object_or_404(User,username=request.user.username)
+    if(request.user.username == "AG") :
```

Comment:
why is this username hardcoded here, this may break in production if the user does not exist or has a different username.

Reply:
In our whole module this is only hard coded due to no entry of dept admin which going to track  the log of assistanship data.Basically i have to give this event access to dept admin only but i have no credential ,So i give to AG  to show the functionality.I can change if required but i need database entry for this.

Reply:
This will trouble you because the username "AG" does not exist in the updated database in the latest_staging. You can confrim the same as well. Infact you can talk to the database team regarding this and add a credential or designation for the same in the staging server which would make it much easier for you to demonstrate.